### ISSUE TITLE
Test coverage for data storers and MQTT handler

### ISSUE THREAD

**Issue by oscgonfer** [fablabbcn/smartcitizen-api#263]
BTW @oscgonfer when this refactor is done and merged, I'd like to start thinking about refactoring and improving the test coverage for the data storers and MQTT handler - they're extremely "enmarañados" at the moment and with quite flaky coverage, which is making me very nervous about changing anything to do with them, and they're a pretty key part of the system. We can discuss next time we see each other!

_Originally posted by @timcowlishaw in https://github.com/fablabbcn/smartcitizen-api/issues/246#issuecomment-1750139111_

**Comment by timcowlishaw** [comment]
Yeah, i'd still like to do this, it worries me. I think it might end up being part of a bigger rewrite, possibly for Kit v3 though!
